common/src/main/java/com/microsoft/identity/common/internal/providers/oauth2/WebViewAuthorizationFragment.java

```diff
@@ -227,9 +239,117 @@ public boolean onTouch(final View view, final MotionEvent event) {
         mWebView.setVisibility(View.INVISIBLE);
+        mWebView.getSettings().setPluginState(WebSettings.PluginState.ON);
         mWebView.setWebViewClient(webViewClient);
+        mWebView.setWebChromeClient(new WebChromeClient() {
+            @Override
+            public void onPermissionRequest(final PermissionRequest request) {
+                // We can only grant or deny permissions for video capture/camera.
+                // To avoid unintentionally granting requests for not defined permissions.
+                if (iSPermissionRequestForCamera(request)) {
+                    Logger.info(methodTag, "Camera request received.");
+                    mCameraPermissionRequest = request;
+                    if (isCameraPermissionGranted()) {
+                        Logger.info(methodTag, "Camera permission already granted.");
+                        acceptCameraRequest();
+                    } else if (shouldShowRequestPermissionRationale(Manifest.permission.CAMERA)) {
+                        Logger.info(methodTag, "Show camera rationale.");
+                        showCameraRationale();
+                    } else {
+                        requestCameraPermissionFromUser();
+                    }
+                }
+            }
+        });
+    }
+
+    /**
+     * Call this method to grant the permission to access the camera resource.
+     * The granted permission is only valid for the current WebView.
+     */
+    private void acceptCameraRequest() {
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
+            final String[] cameraPermission = new String[] {
+                    PermissionRequest.RESOURCE_VIDEO_CAPTURE
+            };
+            if (mCameraPermissionRequest != null) {
+                mCameraPermissionRequest.grant(cameraPermission);
+            }
+        }
+    }
+
+    /**
+     * Call this method to deny the permission to access the camera resource.
+     */
+    private void denyCameraRequest() {
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP &&
+                mCameraPermissionRequest != null) {
+            mCameraPermissionRequest.deny();
+        }
+    }
+
+    /**
+     * Determines whatever if the camera permission has been granted.
+     *
+     * @return true if the camera permission has been granted, false otherwise.
+     */
+    private boolean isCameraPermissionGranted() {
+        return ContextCompat.checkSelfPermission(requireContext(), Manifest.permission.CAMERA)
+                == PackageManager.PERMISSION_GRANTED;
     }
 
+    /**
+     * Determines whatever if the given permission request is for the camera resource.
+     *
+     * @param request The permission request.
+     * @return true if the given permission request is for camera, false otherwise.
+     */
+    private boolean iSPermissionRequestForCamera(final PermissionRequest request) {
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
```

Comment:
Can you add some comments about OS versions checks?

It seems if the OS is not from among that range then we are assuming permission request isn't for camera by hard code returning false

Reply:
I added a comment with more context about this version check. 
```
     * Note: This method is only available on API level 21 or higher.
     * Devices running on lower API levels will not be able to grant or deny camera permission requests.
     * getResources() method is only available on API level 21 or higher.
```
and a warning message
```
        Logger.warn(TAG, "PermissionRequest.getResources() method is not available on API:"
                + Build.VERSION.SDK_INT + ". We cannot determine if the request is for camera.");
```